### PR TITLE
Improved webpage layout & theme so it is easier to find what you need

### PR DESCRIPTION
**Pull request by ahundt**
image

1. Adds the `sphinx-readable-theme` sphinx layout to improve the accessibility and appearance of the documentation. 
2. Reorganizes the rst files so the sidebar displays the information you need to find your way around.

Based on #2477.

[sphinx-readable-theme](sphinx-readable-theme.readthedocs.io/en/latest) contains an example on the package's read the docs page.

to get the package:
`pip install sphinx-readable-theme`

It is also listed in `docs-requirements.txt`

To build:

go to the `docs` folder and type `make html`. 
Make sure you have all necessary packages installed,
everything will appear in the `docs/_build/html` folder.

### CODE REVIEW

**Comment by ahundt** [comment]
Upstream build is failing so I don't think it is my fault...

**Comment by melund** [comment]
Thanks a lot  @ahundt  for the restructuring. I think that is a great idea.

The theme seems a bit boring and not that friendly to me.  But that is only based on [the documentation site](https://sphinx-readable-theme.readthedocs.io/en/latest/). I have haven't see the real thing because I have never been able to build the documentation locally on Windows. If anyone know what this error is about let me know: 

```
snail@sea ~\Documents\xonsh\docs ahundt-sphinx-readable-theme-pypi
$ make html
Running Sphinx v1.6.2

Encoding error:
'charmap' codec can't encode character '\u25cf' in position 28177: character maps to <undefined>
The full traceback has been saved in C:\Users\mel\AppData\Local\Temp\sphinx-err-49f2x36c.log, if you want to report the issue to the developers.
```

@gforsyth Is there a way to preview PRs with [doctr](https://github.com/drdoctr/doctr)?

**Comment by gforsyth** [comment]
Hey @melund -- yes, but there are restrictions. We can only build docs from branches that are in the main repo, since forks don't have access to the deploy key we use to push back to the docs repo.
Then we only build docs from master by default -- that said, you can create a branch and tweak the `.travis.yml` by adding this line:

```python
doctr deploy --sync --no-require-master  --built-docs docs/_build/html "docs-$TRAVIS_BRANCH";
```

which will deploy from a non-master branch to a subfolder in the docs that contains the branch name.

**Comment by ahundt** [comment]
Does master work on Windows? In other words is there a bug in this PR?

The full generated version looks much like the current one just with the fonts and boxes of the new theme and the side bar is much easier to browse, plus there is a separate doc index page and api index page.

**Comment by ahundt** [comment]
Here is a screenshot I took locally:
image

This layout also works on mobile, though the custom html in the xonsh docs folder would need to be fixed.

**Comment by melund** [comment]
@ahundt No. I have never been able to build the documentation on windows. One of the test failures seems to be unrelated, the other error with building documentation is because CI tries to build both the current branch and the last tagged commit. 

That is why it fails with missing  ['cloud_sptheme'](https://travis-ci.org/xonsh/xonsh/jobs/264214068#L1708) . Just add it back in to requirements. It can be removed later.

I tried to push your branch to the main repo, with @gforsyth suggestions to tweak travis. 

So now your work is live here: https://xonsh.github.io/xonsh-docs/docs-ahundt-sphinx-readable-theme-pypi/

I will take a closer look tomorrow. @xonsh/xore needs to join in here.

I am still not sure I like the new theme better. The cloud theme can also be customized in many ways. But I am in favor of improving the structure and the side-bar. I think the guides could use more emphasis in the side-bar. The guides are the most important resource we got at the moment, even though we really need a manual of a sort.

**Comment by astronouth7303** [comment]
Please do the appropriate cleaning up (logo background color, the "fork me on github" corner).

My only comment is that this theme lacks personality. But I don't practice much as a designer.

These theme does seem to help exposing sections on the front page. But I still think we need to dig in and redo a bunch of the current docs.

**Comment by ahundt** [comment]
@astronouth7303 could I get a hand with that github banner fix? I only do minor tweaks to web software once every couple years so I'd have to do some research on how to fix it.

**Comment by astronouth7303** [comment]
@ahundt it should probably go away in favor of a github button or some other design element.

**Comment by ahundt** [comment]
Ok I think I did everything but the new github button, I removed the existing ribbon. I'm going to leave the new github button part to somebody else but maybe [sphinxcontrib-srclinks](https://pypi.python.org/pypi/sphinxcontrib-srclinks) would help. :-)

**Comment by ahundt** [comment]
As for the personality part, I just went for something easy on the eyes and on the browsing. I'd suggest adding some cool demo gifs like you can find in the [visual studio code website](https://code.visualstudio.com/blogs/2017/06/20/great-looking-editor-roundup).

Perhaps a UI design expert willing to contribute to xonsh would be the best option. :-)

**Comment by ahundt** [comment]
I updated the top of this page with a new screenshot from my local machine!

**Comment by astronouth7303** [comment]
`sphinxcontrib-srclinks` is good and useful and should be used. But we should also have an obvious link "this is our github project".

**Comment by astronouth7303** [comment]
Such as https://buttons.github.io/

**Comment by melund** [comment]
@ahundt I have updated 
https://xonsh.github.io/xonsh-docs/docs-ahundt-sphinx-readable-theme-pypi/
with your newest commits. 

The current theme feels more friendly to me. But I agree that it could need a restructure, and maybe other layout options and colors. You can find many of the options on the [`cloud_sptheme` page](http://cloud-sptheme.readthedocs.io/en/latest/cloud_theme_test.html). I can also suggest that we add [`sphinxcontrib-fulltoc`](https://sphinxcontrib-fulltoc.readthedocs.io/en/latest/) to get a more structured sidebar. 

To sum up I am still not in favor of the new theme. But we can take a vote among the xore devs if you want.

**Comment by ahundt** [comment]
Ok well it seems people at least approve of the layout changes, so I'll see if I can make a separate PR with just those parts.

**Comment by ahundt** [comment]
#2483 has the layout changes only, hopefully that's less controversial!

**Comment by melund** [comment]
Thanks a lot, @ahundt. I don't know if it really was controversial. I haven't heard other opinions. But let us take #2483 first.

**Comment by ahundt** [comment]
Would you guys be up for some more discussion of a new docs theme?

The reason I ask is I was browsing the docs to learn a little more about xontribs so I started on the main docs page at xon.sh, and while I was reading the page I ended up coming back to the [readable-wide web page](https://xonsh.github.io/xonsh-docs/docs-ahundt-sphinx-readable-theme-pypi/) @melund put up based on my PR. readable-wide is pretty good and would be easy to finish up since a working version is in the history, but I'll go over some other possibilities and research based suggestions on what makes for a choice below. 

xontribs tutorial comparison
------

I think the xontribs page is really good spot to make a comparison, here is my original suggestion as an example:

[current](http://xon.sh/tutorial_xontrib.html)             |  [readable-wide](https://xonsh.github.io/xonsh-docs/docs-ahundt-sphinx-readable-theme-pypi/tutorial_xontrib.html)
:-------------------------:|:-------------------------:
image | image
image | image

Research based evidence
-------------------------

I also found some research based suggestions for design from the 2008 paper ["An empirical investigation of color temperature and gender effects on web aesthetics"](http://dl.acm.org/citation.cfm?id=2835569) by Coursaris et. al., I included their summary below. Bold areas are mine for emphasis:

**Practitioner’s Take away**
The following list summarizes practical take-aways that practitioners can get from this article:
- Choose a Web site’s colors wisely as they will impact visitors’ impressions of its order and creativity.
- The safest split-complementary color schemes (in terms of influencing users’ impressions of a Web site’s aesthetic appeal) are those with a cool primary color (e.g., blue) for the top or global part of the page. Similarly, a cool secondary color is safer than a warm one. It should be noted that context is an overriding factor; in certain cases warm color combinations may result in higher aesthetic appeal.
 - **Using cool [e.g., blue] color schemes will create favorable impressions about the Web site’s design, which in turn may translate in building credibility and trust.**
- Web site designs that appear orderly are more likely to be also perceived as aesthetically pleasing and in turn more usable.
- No gender differences arose in terms of color preference in this context (i.e., hotel Web sites).
- When gauging the appeal of a Web site design, two discriminant dimensions should always be measured, i.e., classical and expressive aesthetics.
- The measurement of classical aesthetics centers on the Web site’s orderly presentation. **Designers and managers should gauge users’ perceptions on the Web site’s clarity, aesthetics, pleasantness (as shown in this study), cleanliness, and symmetry (validated in past studies).**
- Even if designers are interested in producing creative, fascinating Web sites, they would be wise to consider orderly presentation given its very large effect on users’ perceptions of expressive aesthetics.


Other Sphinx Themes
---------------------

The list [awesome-sphinxdoc's themes](https://github.com/yoloseem/awesome-sphinxdoc#themes) has some nice options too, and the [sphinx-bootstrap-theme](http://ryan-roemer.github.io/sphinx-bootstrap-theme/index.html) ([github](https://github.com/ryan-roemer/sphinx-bootstrap-theme)) stood out, check out the [sphinx-bootstrap-theme sidebar demo](http://ryan-roemer.github.io/sphinx-bootstrap-theme/sidebar.html) demo page. 

image

You can [customize the sphinx-bootstrap-theme using bootswatch.com](https://bootswatch.com/):

image
 
Conclusion
-----------

Based on the above readable-wide would be pretty good, especially if the secondary color was changed to blue, but  [sphinx-bootstrap-theme](https://github.com/ryan-roemer/sphinx-bootstrap-theme) ( with a color scheme based on the research based evidence might be better. Also take a look look at the [awesome-sphinxdoc's themes](https://github.com/yoloseem/awesome-sphinxdoc#themes) list. 

Thoughts?

**Comment by melund** [comment]
@ahundt Thanks for the detailed post. To me, the biggest difference in the comparisons between the two themes seems to be the text size. I do agree about the colors thus. I am all for changing those. It is not even the original colors from the [cloud](http://cloud-sptheme.readthedocs.io/en/latest/cloud_theme_test.html)  theme anyway.

**Comment by melund** [comment]
Maybe we can use the bootswatch site to find some good color combinations.
